File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map= {}

        for index in range(len(nums)):
            num = nums[index]
            if num in hash_map:
                return [hash_map[num],index]
            hash_map[target-num] = index

        
        

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> counter;

        for (int i=0;i<nums.size(); i++) {
            int num = nums[i];

            if (counter.count(target - num)) {
                int first_index = counter[target-num];
                int second_index = i;
                return {first_index, second_index};
            }

            counter[num] = i;

        }
        return {};
    }
};